core/chains/evm/client/rpc_client.go

```diff
@@ -803,21 +810,69 @@ func (r *rpcClient) PendingCallContract(ctx context.Context, msg interface{}) (v
 
 	lggr.Debug("RPC call: evmclient.Client#PendingCallContract")
 	start := time.Now()
+	var hex hexutil.Bytes
 	if http != nil {
-		val, err = http.geth.PendingCallContract(ctx, message)
+		err = http.rpc.CallContext(ctx, &hex, "eth_call", toCallArg(message), "pending")
 		err = r.wrapHTTP(err)
 	} else {
-		val, err = ws.geth.PendingCallContract(ctx, message)
+		err = ws.rpc.CallContext(ctx, &hex, "eth_call", toCallArg(message), "pending")
 		err = r.wrapWS(err)
 	}
+	if err == nil {
+		val = hex
+	}
 	duration := time.Since(start)
 
 	r.logResult(lggr, err, duration, r.getRPCDomain(), "PendingCallContract",
 		"val", val,
 	)
 
 	return
+}
 
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+func toBlockNumArg(number *big.Int) string {
+	if number == nil {
+		return "latest"
+	}
+	if number.Sign() >= 0 {
+		return hexutil.EncodeBig(number)
+	}
+	// It's negative.
+	if number.IsInt64() {
+		return rpc.BlockNumber(number.Int64()).String()
+	}
+	// It's negative and large, which is invalid.
+	return fmt.Sprintf("<invalid %d>", number)
+}
+
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+// Modified to include legacy 'data' as well as 'input' in order to support non-compliant servers.
+func toCallArg(msg ethereum.CallMsg) interface{} {
```

Comment:
@jmank88 ethereum recently updated this function to and added extra fields, should we reflect those changes?

https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient.go#L642

Reply:
nvm 13.1.8 doesnot use it,  just need to ensure we update this function if we do modify to new geth version

Reply:
I copied from the version we import to minimize changes